docs/DevGuide/Observers.md

```diff
@@ -0,0 +1,116 @@
+\cond NEVER Distributed under the MIT License.  See LICENSE.txt for details.
+\endcond
+# Observers Infrastructure {#observers_infrastructure_dev_guide}
+
+The observers infrastructure works with two parallel components: a group and a
+nodegroup. We have two types of observations `Reduction` and `Volume` (see the
+enum observers::TypeOfObservation). `Reduction` data is anything that is written
+once per simulation. Some examples of reduction data are integrals or L2 norms
+over the entire domain, integrals or L2 norms over part of the domain, and
+integrals over lower-dimensional surfaces such as apparent horizons or slices
+through the domain. Volume data is anything that has physical extent, such as
+any of the evolved variables (or derived quantities thereof) across all or part
+of the domain, or quantities on lower-dimensional surfaces in the domain
+(e.g. the rest mass density in the xy-plane). Reduction and volume data both use
+the group and nodegroup for actually getting the data to disk, but do so in a
+slightly different manner.
+
+### Reduction Data
+
+Reduction data requires combining information from many or all cores of a
+supercomputer to get a single value. Reductions are tagged by some temporal
+value, which for hyperbolic systems is the time and for elliptic systems some
+combination of linear and non-linear iteration count. The reduction data is
+stored in an object of type `Parallel::ReductionData`, which takes as template
+parameters a series of `Parallel::ReductionDatum`. A `Parallel::ReductionDatum`
+takes as template parameters the type of the data, a binary operator that
+combines two values into one, and optionally an n-ary operator that computes the
+final result using the extra reduction arguments specified as the fourth
+template parameter to `Parallel::ReductionDatum`. See the documentation of
+`Parallel::ReductionDatum` for examples.
+
+At the start of a simulation, every component and event that wants to perform a
+reduction for observation, or will be part of a reduction observation, must
+register with the `observers::Observer` component. The `observers::Observer` is
+a group, which means there is one per core. The registration is used so that the
+`Observer` knows once all data for a specific reduction (both in time and by
+name/ID) has been contributed. Once all the data is collected on the core it is
+copied to the local `observers::ObserverWriter` nodegroup, which keeps track of
+how many of the cores on the node will be contributing to a specific
+observation. Once all the reductions on a node are complete, the final reduction
```

Comment:
-> "Once all the node's data is collected to the nodegroup, it is combined using the binary operator from `Parallel::ReductionDatum`'s second template parameter. Then, in a similar fashion, the final reduction is done to node `0`. ..."

... But I'm not sure if what I wrote above is correct, because I still have an unanswered question after reading your docs: at which stage(s) is the data combined using the binary op? per core? per node (this is what I think is going on from the code?)? once after everything is copied to node 0?

So regardless of whether you like my suggestion for redistributing the content of the abstract sentence above, I think this would be a good thing to clarify.

Reply:
I have expanded this discussion to clarify when things are combined. Please let me know what you think!